### DISCUSSION TITLE
Is passing Images directly from GPU Memory to ocr possible?

### DISCUSSION THREAD

**Discussion by vajos** [PaddlePaddle/PaddleOCR#12863] [Q&A]
## How to Pass GpuMat Directly to `ocr` Function that Expects `(np.ndarray, list, str, bytes)` Type for `img` Parameter?

The Paddle OCR function defined as follows:

```python
def ocr(self, img, det=True, rec=True, cls=True, bin=False, inv=False, alpha_color=(255, 255, 255)):
    # function implementation
```

This function expects the img parameter to be of type (np.ndarray, list, str, bytes).

However, I'm using OpenCV's CUDA implementation, and my image is stored as a GpuMat object. My goal is to avoid transferring the image from GPU to CPU memory, as that would add an unnecessary overhead. The only current solution is to use download and pass this inside.

Is it possible to directly pass a GpuMat object to this ocr function? 

Or is anything like that planned? I'd like to keep track and set up a notification for updates.

**Comment by prajwaltulawe** [comment]
Any updates on this ?

**Comment by xinedison** [comment]
Any suggestion for this problem.I'd like to use gpumat to recognize image numbers.
